eclipse/src/saros/ui/model/roster/RosterContentProvider.java

```diff
@@ -1,216 +1,66 @@
-
-  @Inject private volatile DiscoveryManager discoveryManager;
-
-  private final DiscoveryManagerListener discoveryManagerListener =
-      new DiscoveryManagerListener() {
-        @Override
-        public void featureSupportUpdated(final JID jid, String feature, boolean isSupported) {
-
-          // TODO maybe use display.timerExec to avoid massive refresh calls
-          ViewerUtils.refresh(viewer, true);
-        }
-      };
-
-  private final RosterListener rosterListener =
-      new RosterListener() {
-        @Override
-        public void presenceChanged(Presence presence) {
-          ViewerUtils.refresh(viewer, true);
-
-          final String user = presence.getFrom();
+  @Inject private XMPPContactsService contactsService;
 
-          if (user != null) querySarosSupport(Collections.singletonList(user));
-        }
-
-        @Override
-        public void entriesUpdated(Collection<String> addresses) {
-          ViewerUtils.refresh(viewer, true);
-          querySarosSupport(addresses);
-        }
-
-        @Override
-        public void entriesDeleted(Collection<String> addresses) {
-          ViewerUtils.refresh(viewer, true);
-        }
-
-        @Override
-        public void entriesAdded(Collection<String> addresses) {
-          ViewerUtils.refresh(viewer, true);
-          querySarosSupport(addresses);
-        }
-      };
+  private Viewer viewer;
+  private final IContactsUpdate contactsUpdate =
+      (contact, type) -> ViewerUtils.refresh(viewer, true);
 
   public RosterContentProvider() {
     SarosPluginContext.initComponent(this);
-
-    discoveryManager.addDiscoveryManagerListener(discoveryManagerListener);
+    contactsService.addListener(contactsUpdate);
   }
 
   @Override
   public void inputChanged(Viewer viewer, Object oldInput, Object newInput) {
     this.viewer = viewer;
-
-    if (oldInput instanceof Roster) ((Roster) oldInput).removeRosterListener(rosterListener);
-
-    roster = null;
-
-    if (newInput instanceof Roster) {
-      roster = (Roster) newInput;
-      roster.addRosterListener(rosterListener);
-
-      for (final RosterEntry entry : roster.getEntries())
-        querySarosSupport(Collections.singletonList(entry.getUser()));
-    }
-  }
-
-  @Override
-  public void dispose() {
-    if (roster != null) roster.removeRosterListener(rosterListener);
-
-    discoveryManager.removeDiscoveryManagerListener(discoveryManagerListener);
-
-    roster = null;
-    discoveryManager = null;
   }
 
   /**
-   * Returns {@link RosterGroup}s followed by {@link RosterEntry}s which don't belong to any {@link
-   * RosterGroup}.
+   * Returns ContactGroups followed by Contacts which don't belong to any Group.
+   *
+   * @param unused is unused
    */
   @Override
-  public Object[] getElements(Object inputElement) {
-
-    if (!(inputElement instanceof Roster)) return new Object[0];
-
-    Roster inputRoster = (Roster) inputElement;
-    final List<Object> elements = new ArrayList<Object>();
-
-    /*
```

Comment:
The information is now silently lost somewhere in the network layer. You now have to read code.